### ISSUE TITLE
mqtt2 MTConnect/Probe/ topic name for agent

### ISSUE THREAD

**Issue by robot-ranger** [mtconnect/cppagent#385]
agent device probe topic is prefixed with `Agent.` 

![image](https://github.com/mtconnect/cppagent/assets/93950632/db743097-460e-45b2-b790-fa82c91d4443)

is that intentional? what is `Agent.` in this case? is that unique to the 'agent' device type?

If it is device name - why dont the other devices have topics with the same [name].[uuid] format?

**Comment by wsobel** [comment]
It is the type of Device. It is the same as the node below the Devices collection in the Devices model. BestW(Sent from mobile)On Dec 22, 2023, at 00:55, robot-ranger ***@***.***> wrote:﻿
agent device probe topic is prefixed with Agent.
image.png (view on web)
is that intentional? what is Agent. in this case? is that unique to the 'agent' device type?
If it is device name - why dont the other devices have topics with the same [name].[uuid] format?

**Comment by robot-ranger** [comment]
ok, right. i was expecting to be able to walk the json to Current/Agent by path instead of by name. if that makes any sense.

since the name is `Agent.[uuid]` , and uuid is unknown to the client until connect, in order to find the agent, i have to subscribe to `MTConnect/Current/#` and iteratively inspect each published message for a `DeviceStream` with name == `Agent`

...but the owner/operator of the agent COULD have changed the name...

is the expectation that we do some logical `if topic.startsWith('MTConnect/Current/Agent')`?
This would be abusive on behalf of the owner/operator of the agent, but what if one of the devices' uuid started with 'Agent'? (then the if statement above would be true for more than one device)

**Comment by robot-ranger** [comment]
TLDR - imo, the agent probe and current should be accessible by name-agnostic path to improve the flexibility of client apps.

**Comment by wsobel** [comment]
Not sure what you mean exactly. Pls give an example.

**Comment by robot-ranger** [comment]
I cant exclusively subscribe to the agent because my client app may not know the agent uuid and therefore does not know the topic string to subscribe to. The only way to "find" the agent is to subscribe to the entire `/Probe/+` or `/Current/+` topic and inspect each topic. The client apps may not care about ALL of the devices, maybe only wants data from some adapters, but also has a need to reference the agent.

for maximum inter-operability between client apps and cppagent - I propose that the agent should have fixed and common topic strings without uuid, so that any app or adapter can locate and query the agent without knowing uuid

something like `MTConnect/Current/Agent/` and `MTConnect/Probe/Agent/`

Any app or data consumer that wants to further verify the agent by uuid, it still can.

**Comment by robot-ranger** [comment]
i suppose i could unsubscribe... 🤔

**Comment by wsobel** [comment]
What would you suggest in the case where there is more than one agent per broker?Do we introduce a “node id” concept like SpB where we add the agent id as a prefix for all devices including the agent? (Sent from mobile)On Dec 24, 2023, at 01:31, robot-ranger ***@***.***> wrote:﻿
i suppose i could unsubscribe... 🤔

**Comment by robot-ranger** [comment]
well... i hadnt thought of that; not something we envisioned in our use cases. But that is a good point..

**Comment by wsobel** [comment]
Is this still an issue? Otherwise I will close.
